website/docs/reference/administrative-commands/createIndexes.mdx

```diff
@@ -0,0 +1,165 @@
+---
+sidebar_position: 1
+description: Create one or more indexes on a collection to improve query performance and enable certain query features.
+---
+
+# `createIndexes`
+
+import CodeBlock from '@theme/CodeBlock';
+import CreateIndex from '!!raw-loader!./create-indexes/1-create-simple-index.request.js';
+import CreateUniqueIndex from '!!raw-loader!./create-indexes/2-create-unique-index.request.js';
+import CreateCompoundIndex from '!!raw-loader!./create-indexes/3-create-compound-index.request.js';
+import CreatePartialIndex from '!!raw-loader!./create-indexes/4-create-partial-index.request.js';
+import CreateSparseIndex from '!!raw-loader!./create-indexes/5-create-sparse-index.request.js';
+
+<!-- Category for Command "Administrative, Diagnostic, and Utility Commands" should be organized per folder as present in the compatibility document -->
+
+<!-- Short description of the command and its usage. -->
+
+Create one or more indexes on a given collection to improve query performance or enable certain query features (e.g., full-text, vector, and TTL indexes).
+
+<!-- Syntax structure of the command here if there are no variations, types, or usage examples to command. Otherwise, use syntax inside subcategories-->
+
+<!-- markdownlint-disable MD033 -->
+
+This command accepts a collection name and an array of index specification documents.
+The index specification document specifies the fields to be indexed, the index name, and other options such as uniqueness, partial filter expressions, and expiration times for TTL indexes.
+
+```js
+db.runCommand({
+  createIndexes: '<collection_name>',
+  indexes: [
+    {
+      key: {
+        '<field1>': 1 // ascending
+      },
+      name: '<simple_index_name>'
+    },
```

Comment:
Can we highlight optional fields?  I wondered if this suggestion makes sense :thinking: 

Some syntax contain values, while other contains fillable `<>`, can we make them contain fillable box where we can change values? That was make it necessary to describe field such as `index-type` in the table below.

```suggestion
  indexes: [
    {
      key: {
        '<field1>': '<index-type1>',
        '<fieldN>': '<index-typeN>',  // Optional
      },
      name: '<name>'
      unique: '<boolean>'  // Optional
      partialFilterExpression: {
        '<fieldX>': { '<operator>': '<value>' }
      }  // Optional
      sparse: '<boolean>' // Optional
    },
```

Reply:
We shouldn't have `index-type` since thats not a field and for the purposes of this guide, there are only 2 (`-1`, `1`). It's also explained under `key` in the table. I can add required/optional column to the table.

Reply:
Another thing is that syntax won't really be valid JS compared to the current setup. Explanation on the fields will be covered by the table, including all options available under it. So in the current setup, those values would be `true` for their specific index creation type. Type and optional/required status will also be covered on the table.

Reply:
I'm not sure what is not valid about the syntax? The usage of `'<value>'`, but that's already used right? :thinking: 

At the moment, the syntax repeats `name:` and `key:` 5 times. Hence my suggestion to compact it. Also you can use combinations of compound and unique index, essentially `unique` `partialFilterExpression` and `sparse` are options that can be set to index that's why I suggested a bit more compact form :hugs: 

You are right, we are not describing full text or vector search so ascending and descending is sufficient.